scriptoocloud/src/test/java/edu/asu/diging/scriptoocloud/core/service/impl/DatasetServiceTest.java

```diff
@@ -0,0 +1,188 @@
+package edu.asu.diging.scriptoocloud.core.service.impl;
+
+import edu.asu.diging.scriptoocloud.core.data.DataFileRepository;
+import edu.asu.diging.scriptoocloud.core.data.DatasetRepository;
+import edu.asu.diging.scriptoocloud.core.exceptions.FileSystemStorageException;
+import edu.asu.diging.scriptoocloud.core.model.impl.DataFile;
+import edu.asu.diging.scriptoocloud.core.model.impl.Dataset;
+import edu.asu.diging.simpleusers.core.model.IUser;
+import edu.asu.diging.simpleusers.core.model.impl.SimpleUser;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.*;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.springframework.test.context.support.AnnotationConfigContextLoader;
+
+import java.io.File;
+import java.nio.file.Path;
+import java.time.OffsetDateTime;
+import java.util.*;
+
+/**
+ * Test class for DataFileService
+ *
+ * @author John Coronite
+ */
+
+@ExtendWith(SpringExtension.class)
+@ContextConfiguration(classes = AnnotationConfigContextLoader.class)
+public class DatasetServiceTest {
+
+    private final String username = "username";
+    private final String type = Dataset.class.getSimpleName().toLowerCase(Locale.ROOT);
+    private final int pageSize = 25;
+    private final int pageNumber = 1;
+
+
+    @InjectMocks
+    private DatasetService datasetService;
+
+    @Mock
+    private DataFileRepository dataFileRepository;
+
+    @Mock
+    private DatasetRepository datasetRepository;
+
+    @Spy
+    private FileSystemService fileSystemService;
+
+    @Spy
+    private Dataset dataset;
+
+    @Spy
+    private SimpleUser simpleUser;
```

Comment:
why do you need spies? domain objects shouldn't need that in most cases, and the filesystemservice should be mocked.